### PR TITLE
[W05H03] Add behavior tests 

### PR DESCRIPTION
**Pull request by julian-hartl**
I added basic behavior tests for w05h03. They should cover about all edge cases that were described in the exercise.
I wasn't sure about one thing, though. Should the size of the topic array be decreased, when a topic is deleted? Imo, it should not, because it should also not be increased, when a new topic is added.

### CODE REVIEW

File: w04h02/test/pgdp/megamerge/Task1Tests.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package pgdp.megamerge;
```
Comment (nilsreichardt) [inline w04h02/test/pgdp/megamerge/Task1Tests.java:1]: Hey! Thanks for your PR! Is it correct, that you are adding tests for W04H02?

Comment (julian-hartl) [reply w04h02/test/pgdp/megamerge/Task1Tests.java:1]: That commit was probably added by a rebase on the main branch.

Comment (nilsreichardt) [reply w04h02/test/pgdp/megamerge/Task1Tests.java:1]: Can we remove them from the PR? You can just delete them on the branch `add-tests-for-w05h03` of the fork

File: w05h03/test/pgdp.messenger.behavior/PinguTalkTest.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    long getCurrentUserId() throws NoSuchFieldException, IllegalAccessException {
+        final var field = PinguTalk.class.getDeclaredField("userId");
+        field.setAccessible(true);
+        return field.getLong(null);
+
+    }
+
+
+    long getCurrentTopicId() throws NoSuchFieldException, IllegalAccessException {
+        final var field = PinguTalk.class.getDeclaredField("topicId");
```
Comment (HendrikHuebner) [inline w05h03/test/pgdp.messenger.behavior/PinguTalkTest.java:24]:
```suggestion
        final var field = PinguTalk.class.getDeclaredField("topicID");
```

File: w05h03/test/pgdp.messenger.behavior/PinguTalkTest.java [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static pgdp.messenger.unit.TestUtils.getTestUser;
+
+public class PinguTalkTest {
+
+
+    long getCurrentUserId() throws NoSuchFieldException, IllegalAccessException {
+        final var field = PinguTalk.class.getDeclaredField("userId");
```
Comment (HendrikHuebner) [inline w05h03/test/pgdp.messenger.behavior/PinguTalkTest.java:16]:
```suggestion
        final var field = PinguTalk.class.getDeclaredField("userID");
```

**Review by HendrikHuebner — CHANGES_REQUESTED** [review]
LGTM except for the wrong strings
